### PR TITLE
feat(FX-3115): delete saved search alert

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3115]

### Description
* As an admin with the saved search M2 feature flag enabled,
* If I am viewing a saved search "edit" screen,
* I can tap the "delete alert" button
* And be navigated back to the previous screen
* Where I will no longer see the saved search I removed

#### iOS
https://user-images.githubusercontent.com/3513494/129026596-108efccf-c9fd-46ec-97df-41d60eeedd32.mp4

#### Android
https://user-images.githubusercontent.com/3513494/128887862-98bcbcac-dd76-42ae-b9a7-5c83db54270e.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- A user can use "Remove" button to disable a saved search - dzmitry tratsiak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3115]: https://artsyproduct.atlassian.net/browse/FX-3115

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 46d1c87d4bc8bff3ebc21826a4eef85f5255b617
</p>

File: src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx [hunk 1]
```diff
@@ -35,11 +38,31 @@ export const SavedSearchAlertForm: React.FC<SavedSearchAlertFormProps> = (props)
     },
   })
 
+  const onDelete = async () => {
+    try {
+      await deleteSavedSearchMutation(savedSearchAlertId!)
+      onDeleteComplete?.()
+    } catch (error) {
+      console.error(error)
```
Comment (iskounen) [inline src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:46]: **question:** What is the user experience if this mutation fails on the backend?

File: src/lib/Scenes/SavedSearchAlert/__tests__/SavedSearchAlertForm-tests.tsx [hunk 2]
```diff
@@ -83,6 +87,33 @@ describe("Saved search alert form", () => {
...
+        savedSearchAlertId="savedSearchAlertId"
+        onDeleteComplete={onDeletePressMock}
+      />
+    )
+
+    fireEvent.press(getByTestId("delete-alert-button"))
+
+    // Click confirm button
+    // @ts-ignore
+    spyAlert.mock.calls[0][2][1].onPress()
```
Comment (iskounen) [inline src/lib/Scenes/SavedSearchAlert/__tests__/SavedSearchAlertForm-tests.tsx:105]: **suggestion:** I think a comment about what `[0][2][1]` represents may help developers in the future.

**Review by iskounen — COMMENTED** [review]
These changes look good to me. I have a question about error handling and a suggestion about documentation, but otherwise the demos look great!

File: src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:46]: I asked a similar question for update mutation and got [this answer](https://artsy.slack.com/archives/C9SATFLUU/p1628253051179100)

Comment (iskounen) [reply src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:46]: Thanks for asking the team about that! I agree that it's not the most urgent scenario to consider for an unexpected situation.

File: src/lib/Scenes/SavedSearchAlert/__tests__/SavedSearchAlertForm-tests.tsx (hunk 2 above)
Comment (iskounen) [reply src/lib/Scenes/SavedSearchAlert/__tests__/SavedSearchAlertForm-tests.tsx:120]: Thank you 🙌

**Review by iskounen — APPROVED** [review]

**Pull request merged**